pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -464,15 +464,15 @@ fi
 # current directory (for "build" and "test").
 if [ -z "$rollback" ]; then
     echo "building the system configuration..." >&2
-    if [ "$action" = switch -o "$action" = boot ]; then
+    if [[ "$action" = switch ]] || [[ "$action" = boot ]]; then
         if [[ -z $flake ]]; then
             pathToConfig="$(nixBuild '<nixpkgs/nixos>' --no-out-link -A system "${extraBuildFlags[@]}")"
         else
             pathToConfig="$(nixFlakeBuild "$flake#$flakeAttr.config.system.build.toplevel" "${extraBuildFlags[@]}" "${lockFlags[@]}")"
         fi
         copyToTarget "$pathToConfig"
         targetHostCmd nix-env -p "$profile" --set "$pathToConfig"
-    elif [ "$action" = test -o "$action" = build -o "$action" = dry-build -o "$action" = dry-activate ]; then
+    elif [[ "$action" = test ]] || [[ "$action" = build ]] || [[ "$action" = dry-build ]] || [[ "$action" = dry-activate ]]; then
```

Comment:
```suggestion
    elif [[ "$action" = test || "$action" = build || "$action" = dry-build || "$action" = dry-activate ]]; then
```
I think we should at least remove the optional ]] [[ in between. It makes the code shorter and for me personally way easier to understand. I would also be fine with keeping the -o to reduce the diff.